### ISSUE TITLE
Code Lab Feedback

### ISSUE THREAD

**Issue by michaelhenry91** [michaelhenry91/codelab-html-intro#1]
Hey @chrisvfritz, can you take a look at this? It's [hosted here](http://messy-meat.surge.sh/) and meets the following [project](http://lansingcodelab.com/lessons/html-intro/1) criteria:
- [x] Use at least one `h1` element.
- [x] Use at least one `a` element.
- [x] Use at least one `p` element.
- [x] Use at least one `img` element.
- [x] Use at least one `ul` element.
- [x] Use at least one `li` element.
- [x] When you paste the address of your website into [the W3C validator](https://validator.w3.org/), it returns no errors (warnings are OK).
- [x] This is the last time I'll mention it, but once again, your project should be on GitHub and hosted online (I recommend using surge again).

**Comment by chrisvfritz** [comment]
The use of all the elements here looks very good and I thoroughly enjoyed the cheeky descriptions of Sean Connery's movie roles! :balloon: :tada: 

I do see a few areas for further improvement though and want to take this opportunity to dive into these fundamentals in more detail.

### Empty `div` element

Empty elements [like this `div`](https://github.com/michaelhenry91/codelab-html-intro/blob/master/index.html#L20) should generally be avoided, since they add unnecessary clutter to the file. Was there something specific you were trying to achieve with it or was it simply left in by accident?

### Consistent spacing at the beginning and end of elements

It's considered good practice to have the same amount of empty lines at the beginning of elements as you have at the end of elements. In [this `head`](https://github.com/michaelhenry91/codelab-html-intro/blob/master/index.html#L4-L5) for example, you have 2 empty lines at the beginning of the element and 0 at the end of the element. Most likely in this case, you want to delete the 2 empty lines at the beginning.

Then in your body, there is 1 empty line [at the beginning](https://github.com/michaelhenry91/codelab-html-intro/blob/master/index.html#L9), but 2 empty lines [at the end](https://github.com/michaelhenry91/codelab-html-intro/blob/master/index.html#L44-L45). In this case, removing the extra line at the end would resolve the issue.

### General indentation rules

The indentation can be further improved here. Here are some general indentation rules for HTML:
- Indentation should always start at 0
- Indentation sizing should be consistent (if using 2 spaces, every indentation should be 2 spaces; if using 4 spaces, every indentation should be 4 spaces)
- Siblings should always be indented at the same level
- Code inside of other code should always be indented (with possible exceptions in HTML for `<head>`, and `<body>` elements, which may be indented at the same level as the `<html>` element)

The main violations of these rules that I see are:
- elements indented directly after a heading element, even though those items are siblings of the heading element, rather than children inside of it (ex: they are not between the `<h1>` and `</h1>` tags)
- a mix of 2-space indentation and 4-space indentation, rather than the consistent use of one or the other

As a sort of training wheels while you're still getting used to indentation, Atom has a tool you can use to automatically reindent the currently file. First, you'll want to select all the text in the document with `Cmd` + `A`, then select from the menu: `Edit` > `Lines` > `Auto Indent`. Give that a try!

---

Try to make these changes, then repeat the steps of adding, committing, and pushing with git. After you've done this, you can respond to this issue and let me know that you've updated the project. If you have any questions or something doesn't quite make sense, please don't hesitate to let me know and I'll be happy to clarify. :smiley:

**Comment by michaelhenry91** [comment]
Hi Chris, 

Per our in-class discussion, I changed the accidental div tag with a br tag to create the line break I was looking for.  

I also removed the extra space at the top and bottom of the file, although when I command+s to save it creates an extra line at the very bottom of the file.

I then used auto-ident to fix the rest of the indentation.  Here is the new link: http://weary-rose.surge.sh/

**Comment by chrisvfritz** [comment]
Almost perfect now! Just one more note for improvement before shipping this:
- `br`and a few other elements are actually self-closing (meaning they don't need a closing tag), so your `<br></br>` should actually be shortened to just `<br>`

**Comment by michaelhenry91** [comment]
Makes sense!  Here is the link without the second "br" tag: [medical-way.surge.sh](url)

**Comment by chrisvfritz** [comment]
Beautiful! :tada: :shipit:
